discovery/data_sharing/v0.1/data_sharing.md

```diff
@@ -0,0 +1,297 @@
+# "Fediscovery": Data Sharing Specification
+
+## Overview
+
+Specification identifier: `data_sharing`
+
+In order to be able to present results, discovery providers need to be
+able to request content from connected fediverse servers. Two different
+concerns need to be distinguished: Retrieving new content as it is
+created and retrieving historic content to backfill search indexes.
+
+## Restrictions On Content Being Shared
+
+A single FASP will never be used by all fediverse servers. To ensure it
+still gets a broad view not limited to a small number of servers,
+fediverse servers MUST share not only local but also remote content with
+the FASP.
+
+Since this may result in multiple servers sharing the same content with
+a single FASP, FASPs MUST deduplicate content.
+
+As a single server is not authoritative for all the content shared and
+to save bandwidth only URIs of content are actually exchanged with the
+FASP. The FASP is then responsible for retrieving the content and
+working with it. The exact details of this differ between capabilties
+and are described in the respective sections.
+
+A fediverse server MUST make sure it only shares content that it is
+actually allowed to share. It MUST NOT share any content that is not
+public. In the case of account and post data it MUST make sure to only
+share content where the creator has opted in to discovery.
+
+The FASP in turn MUST also ensure that creators have opted in to
+discovery before storing and indexing content.
+
+Both parties MUST use the mechanism described in
+[FEP-5feb](https://codeberg.org/fediverse/fep/src/branch/main/fep/5feb/fep-5feb.md)
+to determine if a creator has opted in to discovery.
+
+For account data, both parties MUST ensure the `discoverable` flag is
+set to `true`.
+
+## Subscribing To, Requesting And Receiving Content
+
+### Managing Subscriptions (FASP => Fediverse Server)
+
+In order to subscribe to new content, FASPs can subscribe to events
+by making a `POST` call to the `/data_sharing/v0/event_subscriptions`
+endpoint on the fediverse server.
+
+Example call:
+
+```http
+POST /data_sharing/v0/event_subscriptions
+```
+
+The request body MUST contain a JSON object defining what events to
+subscribe to. This object MUST contain the keys `objectType` and
+`eventType`. It MAY contain the keys `maxBatchSize` and/or
+`threshold`. The keys are defined as follows:
+
+* `category`: One of `content`, `account`. This is the category of
+  objects the FASP is interested in.
```

Comment:
Is this `objectType`? i.e., Actor, Post, Media, etc?

Reply:
It was at some point, but then we decided against it (hence the change from `objectType` to `category`). At this point we do not expect FASPs to handle different types of content very differently, so it does not seem to make much sense having separate subscriptions for them.

This is also more flexible as the fediverse software can decide what it considers "content" worth indexing and the FASP can decide if and how to handle different types.